### DISCUSSION TITLE
Логи запросов

### DISCUSSION THREAD

**Discussion by melkorsan** [totumonline/totum-ru-issues-and-discussions#487] [Как это сделать?] [answered]
Здравствуйте!
Подскажите, как настроить поступление web-хуков. Remotes настроил, из админки post, вставленный вручную отрабатывается. Если обращаться через браузер - ответ приходит положительный, а направляю хуки из стороннего узла - никакой информации.
Может ли влиять на процесс отсутствие авторизации? В Remotes она не задана.

![Снимок экрана от 2023-12-10 18-40-33](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/153440744/df13bbaf-d5b2-4314-b296-f2259d408d5b)

Пример запроса:
connection | close
accept-encoding | gzip, compress, deflate, br
content-length | 178
user-agent | axios/1.6.2
content-type | application/json
accept | application/json, text/plain, */*

{
  "callerNumber": "79587980186",
  "did": "79958883534",
  "accountId": 113075,
  "vpbxId": "15936",
  "callId": "98-1702203641.27553372",
  "timestamp": 1702203641,
  "eventType": "VpbxInboundCallStart"
}

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Когда вы отправляете application/json он приходит не в POST. POST только для `application/x-www-form-urlencoded` или `multipart/form-data`.

У вас данные придут в переменную `$#input`. И они придут как текст. Те их надо будет распаковать ф. `jsonExtract`.

**Reply by ttm-alexeypolunin** [reply]
Вот я на скрине вижу, что они не распакованы.

Еще что может быть — это запрос с `/` и без в пути запроса. Те запрос должен идти на адрес, который вы видите в панели по правой кнопке мыши, когда кликаете на поле с `name` ремоута.

**Comment by melkorsan** [comment]
Проверяю с переменной $#input - то же самое. Вопрос по распаковке - разве распаковка обязательна для добавления в поле? Нельзя все сообщение добавить целиком?
При этом посылаю сходное сообщение с альтернативного узла - добавляется без распаковки, хотя вроде тот же application/json.
POST /Remotes/test_api
Host: qweez.ru
Content-Type:application/json
{
"timestamp": 1702388696,
"eventType": "VpbxInboundCallStart"
}
А аналогичное сообщение:
POST /Remotes/test_api
Host: qweez.ru
Content-Type:application/json
{
"timestamp": 1702388696,
"eventType": "VpbxInboundCallStart"
}
Не добавляется.
Есть возможность как-то проверить, поступает ли оно вообще в систему?

**Reply by ttm-alexeypolunin** [reply]
Не, если вы целиком хотите как текст положить, то нет — можно не распаковывать.

Вот я запросил этот Ремоут из postman — ответ есть.

<img width="506" alt="Снимок экрана 2023-12-12 в 16 35 26" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/a0a71f1a-79c2-45fe-a895-c02f8d138861">

Если система, с которой вы спрашиваете может показать ответ, то вместо succses/error поставьте возврат значения, например добавленного id или вообще всей переменной

<img width="674" alt="Снимок экрана 2023-12-12 в 17 02 30" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/bcda25b2-fe76-439c-b21a-238a34529894">


```
=: $#input
```

или

```
=: $#post[timestamp]
```

или

```
a1=: insert(....; inserts: "id_added_row")

a2=: $#id_added_row
```
